tests/ArmTemplates.Tests/Extractor/Scenarios/ParametersExtractorTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Commands.Executors;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Builders;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.EntityExtractors;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Extractor.Abstractions;
+using Xunit;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Extractor.Scenarios
+{
+    [Trait("Category", "Parameters Extraction")]
+    public class ParametersExtractorTests : ExtractorMockerWithOutputTestsBase
+    {
+        public ParametersExtractorTests() : base("parameters-tests")
+        {
+        }
+
+        ExtractorExecutor GetExtractorInstance(ExtractorParameters extractorParameters) 
+        {
+            var parametersExtractor = new ParametersExtractor(new TemplateBuilder(), null);
+
+            var loggerExtractor = new LoggerExtractor(
+                this.GetTestLogger<LoggerExtractor>(),
+                new TemplateBuilder(),
+                null,
+                null);
+
+            var extractorExecutor = ExtractorExecutor.BuildExtractorExecutor(
+                this.GetTestLogger<ExtractorExecutor>(),
+                parametersExtractor: parametersExtractor,
+                loggerExtractor: loggerExtractor);
+
+            extractorExecutor.SetExtractorParameters(extractorParameters);
+
+            return extractorExecutor;
+        }
+
+        [Fact]
+        public async Task GenerateParametersTemplates_ProperlyLaysTheInformation()
+        {
+            // arrange
+            var currentTestDirectory = Path.Combine(this.OutputDirectory, nameof(GenerateParametersTemplates_ProperlyLaysTheInformation));
+
+            var extractorConfig = this.GetMockedExtractorConsoleAppConfiguration(
+                splitApis: false,
+                apiVersionSetName: string.Empty,
+                multipleApiNames: string.Empty,
+                includeAllRevisions: false);
+            
+            extractorConfig.ParamNamedValue = "false";
```

Comment:
Thanks for suggestion!

Upon real-world discussion, decided to make another method in `ExtractorMockerWithOutputTestsBase`:
- `CreateDefaultExtractorConsoleAppConfiguration()`
- `CreateDefaultExtractorParameters()`

That will allow both to simulate the user experience with providing "weird" arguments like `"true"` directly in consoleAppConfiguration if needed, and to create an `ExtractorParameters` instance with ease. This also allows to not be dependant on single number of "mocked-default" input parameters specified

Reply:
Also if you feel you have enough time to look through other test-classes and to change the behaviour there, feel free to do it!

Reply:
Method for getting default parameters is introduced, however I would like to suggest making changes for other tests in separate PR, or during changes in particular tests, due to newly introduced CreateDefaultExtractorParameters method has minimal amount of default values for parameters the rest is null. 
Please let me know what you think about this :)